src/library/Yi/Buffer/HighLevel.hs

```diff
@@ -282,11 +282,16 @@ prevNParagraphs n = replicateM_ n $ moveB unitEmacsParagraph Backward
 -- @goUnmatchedB Forward '{' '}'@
 -- Move to the next unmatched '}'
 goUnmatchedB :: Direction -> Char -> Char -> BufferM ()
-goUnmatchedB dir cStart' cStop' = stepB >> readB >>= go (0::Int)
-  where go opened c | c == cStop && opened == 0 = return ()
-                    | c == cStop                = stepB >> readB >>= go (opened-1)
-                    | c == cStart               = stepB >> readB >>= go (opened+1)
-                    | otherwise                 = stepB >> readB >>= go opened
+goUnmatchedB dir cStart' cStop' = getLineAndCol >>= \position ->
+  stepB >> readB >>= go position (0::Int)
+  where go pos opened c
+           | c == cStop && opened == 0 = return ()
+           | c == cStop       = stepB >> readB >>= go pos (opened-1)
+           | c == cStart      = stepB >> readB >>= go pos (opened+1)
```

Comment:
`c == cStart` branch doesn't check for EOF or SOF, wouldn't that fall into infinite loop?